packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -803,9 +819,19 @@ class _LargeTitleNavigationBarSliverDelegate
       middleVisible: alwaysShowMiddle ? null : !showLargeTitle,
     );
 
+    final Color expandedBackgroundColor = expandedTransparent ? backgroundColor.withOpacity(0) : backgroundColor;
+
     final Widget navBar = _wrapWithBackground(
-      border: border,
-      backgroundColor: CupertinoDynamicColor.resolve(backgroundColor, context),
+      border: expandedTransparent
+          ? Border.lerp(null, border, expandedState)
+          : border,
+      backgroundColor: expandedTransparent
+          ? Color.lerp(
+              CupertinoDynamicColor.resolve(expandedBackgroundColor, context),
+              CupertinoDynamicColor.resolve(backgroundColor, context),
+              expandedState,
+            )!
```

Comment:
My eyes may be playing tricks on me, but it looks like the background transition on the iPhone is a bit smoother than the one made by this PR. It also looks like the border may be on a different animation curve. It comes in somewhat suddenly at the end, when meanwhile the color had already started it's transformation. I wonder if their respective `lerp`s should be looking at different double values? Maybe the one for color starts earlier? Or maybe a curve can be applied to it to start appearing sooner?

Reply:
Hi Mitchell, thanks for your feedback.
I've updated the expandedState calculations, it should be much smoother now.

Reply:
Demo: 


https://github.com/flutter/flutter/assets/66907532/7c7f4135-219e-4c07-b33e-4a73dc8a8451